resources/ts/Components/ContactsTableContent.tsx

```diff
@@ -0,0 +1,146 @@
+import React, { ComponentProps } from "react";
+import { ContactsTableProps } from "./ContactsTable";
+import { classNames } from "../utils/classNames";
+import { ButtonIcon } from "./ButtonIcon";
+import { faPencil, faTrash } from "@fortawesome/free-solid-svg-icons";
+import { useTranslation } from "react-i18next";
+import { ContactFormData } from "./Integrations/Detail/ContactInfo";
+import { Contact } from "../Pages/Integrations/Index";
+type Props = { desktop?: boolean; mobile?: boolean } & ContactsTableProps &
+  ComponentProps<"div">;
+
+export const ContactsTableContent = ({
+  desktop,
+  mobile,
+  data,
+  onEdit,
+  onDelete,
+  onPreview,
+  functionalId,
+  technicalId,
+}: Props) => {
+  const { t } = useTranslation();
+  const contactTypes = [
+    { label: "functional", id: functionalId },
+    { label: "technical", id: technicalId },
+  ];
+
+  return (
+    <tbody>
+      {contactTypes.map((type) => (
+        <tr
+          className={classNames(
+            "bg-white border-b",
+            desktop && "hover:bg-publiq-gray-light"
+          )}
+          key={type.id}
+        >
+          <td
+            className={classNames(
+              desktop && "w-2/4",
+              mobile && "px-2 py-4 w-full"
+            )}
+            onClick={
+              mobile
+                ? () => {
+                    onPreview(true);
+                    onEdit(type.id);
+                  }
+                : undefined
+            }
+          >
+            <div
+              className={classNames(
+                desktop && "px-6 pt-4 font-semibold",
+                mobile &&
+                  "text-publiq-blue-dark flex gap-2 items-center underline"
+              )}
+            >
+              {
+                (data[type.label as keyof ContactFormData] as Contact)
```

Comment:
We can probably search the correct contact in the mapping of contactTypes

Something like this:

```ts
{contactTypes.map((type) => {
    const contact = data[type.label];
    
    return (
        <tr
```

Also you can change
```ts
const contactTypes = [
    { label: "functional", id: functionalId },
    { label: "technical", id: technicalId },
  ];
```

To

```ts
const contactTypes = [
    { label: "functional", id: functionalId },
    { label: "technical", id: technicalId },
  ] as const;
```

You won't need the `as keyof ContactFormData`